src/biotite/structure/charges.py

```diff
@@ -277,15 +415,64 @@ def partial_charges(atom_array, iteration_step_num=6, charges=None):
                 UserWarning
             )
 
-    bonds, _ = atom_array.bonds.get_all_bonds()
+    elements = atom_array.element
+    bonds, types = atom_array.bonds.get_all_bonds()
     amount_of_binding_partners = np.count_nonzero(bonds != -1, axis=1)
+    # The maximum of a given row of the `types` array must be determined
+    # as this value reveals the hybridisation state
+    try:
+        bond_types = np.amax(types, axis=1)
+        zero_indices_in_first_dim = np.unique(
+            np.nonzero(types == 0)[0]
+        )
+        bond_types[zero_indices_in_first_dim] = 0
+        # As nitrogen can be involved in aromatic systems having both
```

Comment:
I do not quite understand , why special handling for nitrogen is necessary. Shouldn't  `bond_types = np.amax(types, axis=1)` already give the correct hybridization?

Reply:
I added a comment explaining the need to proceed this way.